### PR TITLE
Refactor linkages

### PR DESCRIPTION
**Pull request by netochaves**
#### What does this PR do?
Refactoring of linkages reducer
#### Where should the reviewer start?
`src/reducers/linkages.js`
#### What testing has been done on this PR?
unit test
#### Any background context you want to provide?
N/A
#### What are the relevant issues?
#44 
#### Screenshots (if appropriate)
N/A

### CODE REVIEW

File: src/reducers/linkages.js [hunk 1]
```diff
@@ -5,26 +5,25 @@ import {
   REMOVE_LINKAGE,
 } from 'actions';
 
+import { dissoc, path, assoc } from 'ramda';
+
+const getLinkagesFromAction = action =>
```
Comment (nolleto) [inline src/reducers/linkages.js]: In `ramda` all functions are curried so, you can simplify this method like this:

```js
const getLinkagesFromAction = 
  path(['payload', 'state', 'network', 'linkages']);
```

File: src/reducers/linkages.js [hunk 2]
```diff
@@ -5,26 +5,25 @@ import {
...
   switch (action.type) {
     case NEW_NETWORK:
       return {};
-    case ADD_LINKAGE:
-      return {
-        ...state,
-        [Date.now()]: action.payload.linkage,
-      };
+    case ADD_LINKAGE: {
+      const { linkage } = action.payload;
```
Comment (nolleto) [inline src/reducers/linkages.js]: You could create a function for this:
```js
import { path } from 'ramda';

const getLinkageFromAction = path(['payload', 'linkage'])
...
return assoc(Date.now(), getLinkageFromAction(action), state);
```

File: src/reducers/linkages.js [hunk 3]
```diff
@@ -5,26 +5,25 @@ import {
...
+      return assoc(Date.now(), linkage, state);
+    }
     case REMOVE_LINKAGE: {
       const { id } = action.payload;
```
Comment (nolleto) [inline src/reducers/linkages.js]: You could create a function for this:
```js
import { path } from 'ramda';

const getIdFromAction = path(['payload', 'id']);
...
return dissoc(getIdFromAction(action), state);

File: src/reducers/linkages.js [hunk 4]
```diff
@@ -5,26 +5,25 @@ import {
...
-
-      return newState;
-    }
-    case LOAD_NETWORK: {
-      const { linkages } = action.payload.state.network;
-      return linkages || {};
+      return dissoc(id, state);
     }
+    case LOAD_NETWORK:
+      return getLinkagesFromAction(action) || {};
```
Comment (nolleto) [inline src/reducers/linkages.js]: You can use `defaultTo` here and [pipe](https://ramdajs.com/docs/#pipe) to combine these function:

```js
const getLinkagesFromAction = 
  path(['payload', 'state', 'network', 'linkages']);

// you can choose a better name if you wanna...
const getLinkagesOrDefaultFromAction = pipe(getLinkagesFromAction, defaultTo({}))
...
return getLinkagesOrDefaultFromAction(action);
```

Just to be clear, I can rewrite this function like that:
```js
const getLinkagesFromAction = 
  path(['payload', 'state', 'network', 'linkages']);
const defaultToEmptyObject = defaultTo({})

const getLinkagesOrDefaultFromAction = action => {
  const linkages = getLinkagesFromAction(action);
  
  return defaultToEmptyObject(linkages)
}
```
😄

**Comment by netochaves** [comment]
Thanks for your review @nolleto

**Pull request merged**